From 717ff7a3ae3e012ce8a2e83230acbab4232e7d67 Mon Sep 17 00:00:00 2001
From: Jim MacArthur <jim.macarthur@codethink.co.uk>
Date: Wed, 11 Nov 2015 18:50:19 +0000
Subject: [PATCH 36/47] Allow redefinition of types for procedures

---
 gcc/fortran/symbol.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/fortran/symbol.c b/gcc/fortran/symbol.c
index 28acc75..f1b7f16 100644
--- a/gcc/fortran/symbol.c
+++ b/gcc/fortran/symbol.c
@@ -1723,7 +1723,7 @@ gfc_add_type (gfc_symbol *sym, gfc_typespec *ts, locus *where)
 	    /* Ignore temporaries and class/procedure names */
 	    return FAILURE;
 	  }
-	  if (gfc_compare_types(&sym->ts, ts) && (flavor == FL_UNKNOWN || flavor == FL_VARIABLE))
+	  if (gfc_compare_types(&sym->ts, ts) && (flavor == FL_UNKNOWN || flavor == FL_VARIABLE || flavor == FL_PROCEDURE))
 	    {
 	      return gfc_notify_std (GFC_STD_LEGACY,
 				     "Symbol '%s' at %L already has basic type of %s", sym->name,
-- 
1.9.1

